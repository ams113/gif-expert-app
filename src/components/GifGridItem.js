import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( { title = '', url = '' } ) => {

    const titleValid = title.trim() || 'Without title';

    return (
        <article className="card animate__animated animate__fadeIn">
            <div className="card-content">
                <img src={ url } alt={ titleValid } />
                <p>{ titleValid }</p>
            </div>
        </article>
    )
}


GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired,
  };