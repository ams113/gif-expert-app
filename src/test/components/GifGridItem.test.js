import React from 'react';
import { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';

describe('Testing components: GifGridItem', () => {
    
    const title = 'fakeText';
    const url ='https://fake.dev/picture/fake.giff';
    
    const wrapper = shallow( <GifGridItem title={ title } url={ url } /> );

    test('should renders correctly', () => {

        expect( wrapper ).toMatchSnapshot();
    });

    test('should tag <p></p> with title ', () => {

        expect( wrapper.find('p').text().trim() ).toBe( title );
    });
    
    test('should contain the [src] and [alt] attributes of the image as the props', () => {

        const img = wrapper.find('img');
        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });

    test('should contain animate__fadeIn class', () => {

        const article = wrapper.find('article');
        const className = article.prop('className');

        expect( className.includes('animate__fadeIn') ).toBe(true);

    });

    test('should show a default title if the props is empty', () => {

        const wrapper = shallow( <GifGridItem title={ '' } url={ url } /> );
        expect( wrapper.find('p').text().trim() ).toBe( 'Without title' );

    });
    

});
