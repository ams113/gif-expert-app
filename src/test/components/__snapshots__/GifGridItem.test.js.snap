// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing components: GifGridItem should renders correctly 1`] = `
<article
  className="card animate__animated animate__fadeIn"
>
  <div
    className="card-content"
  >
    <img
      alt="fakeText"
      src="https://fake.dev/picture/fake.giff"
    />
    <p>
      fakeText
    </p>
  </div>
</article>
`;
